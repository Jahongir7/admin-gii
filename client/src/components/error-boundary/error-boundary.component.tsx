import React from 'react';

export class ErrorBoundary extends React.Component<any, { hasError: boolean }> {
  constructor(props: any) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  componentDidCatch(error: any) {
    console.log(
      '%c Caught an error!',
      'padding: 5px; background: #e5e0d4; color: #ff1f1f;'
    );
    console.log(error);
  }

  render() {
    return this.state.hasError ? (
      <h1 {...this.props}>Something is wrong</h1>
    ) : (
      this.props.children
    );
  }
}
