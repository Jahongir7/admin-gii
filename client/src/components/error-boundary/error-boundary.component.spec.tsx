import { customRender } from '@test-utils';
import { screen } from '@testing-library/react';
import { ErrorBoundary } from './error-boundary.component';

describe('Common Context', () => {
  test('Should render ErrorBoundary component', () => {
    const ThrowError = () => {
      throw new Error('Error');
    };
    const { asFragment } = customRender(
      <ErrorBoundary data-testid="error-component">
        <ThrowError />
      </ErrorBoundary>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('error-component')).toBeInTheDocument();
  });
});