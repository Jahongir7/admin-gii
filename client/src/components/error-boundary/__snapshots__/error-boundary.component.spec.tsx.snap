// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common Context Should render ErrorBoundary component 1`] = `
<DocumentFragment>
  <h1
    data-testid="error-component"
  >
    Something is wrong
  </h1>
</DocumentFragment>
`;
